### PR TITLE
Feature/tinder in card

### PR DESCRIPTION
**Pull request by MagdaITechArt**

### CODE REVIEW

**Review by wmosienski — APPROVED** [review]
git good

File: client/src/_components/CustomTinderCardStack/CustomTinderCardStack.jsx [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            const index = offers.map(offer => offer.id).indexOf(toBeRemoved);// Find the index of which to make the reference to
+            alreadyRemoved.push(toBeRemoved); // Make sure the next card gets removed next time if this card do not have time to exit the screen
+            childRefs[index].current.swipe(dir); // Swipe the card!
+        }
+    }
+
+    return (
+        <>
+            <h3 key={lastDirection} className={styles.infoText}>
+                {!lastDirection ? intl.formatMessage({ id: 'customTinderCardStack.swipeToStart' }) :
```
Comment (solleOlle) [inline client/src/_components/CustomTinderCardStack/CustomTinderCardStack.jsx:152]: too complex. this logic shouldn't in the template

File: client/src/_components/CustomTinderCardStack/CustomTinderCardStack.jsx [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+            const toBeRemoved = cardsLeft[cardsLeft.length - 1].id; // Find the card object to be removed
+            const index = offers.map(offer => offer.id).indexOf(toBeRemoved);// Find the index of which to make the reference to
+            alreadyRemoved.push(toBeRemoved); // Make sure the next card gets removed next time if this card do not have time to exit the screen
+            childRefs[index].current.swipe(dir); // Swipe the card!
+        }
+    }
+
+    return (
+        <>
+            <h3 key={lastDirection} className={styles.infoText}>
```
Comment (solleOlle) [inline client/src/_components/CustomTinderCardStack/CustomTinderCardStack.jsx:151]: why `key`?

File: client/src/_components/CustomTinderCardStack/CustomTinderCardStack.jsx [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+            '\n' +
+            '\n' +
+            'Where does it come from?\n'
+    },
+]
+
+const alreadyRemoved = [];
+let offersState = offers; // This fixes issues with updating characters state forcing it to use the current state and not the state that was active when the card was created.
+
+function CustomTinderCardStack() {
```
Comment (solleOlle) [inline client/src/_components/CustomTinderCardStack/CustomTinderCardStack.jsx:120]: why custom? :)

File: .idea/vcs.xml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (solleOlle) [inline .idea/vcs.xml:1]: `.idea` shouldn't be here. put it in the `.gitignore` please

File: client/src/TinderInPage/TinderInPage.jsx [hunk 5]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import {useIntl} from "react-intl";
+import CustomTinderCardStack from "../_components/CustomTinderCardStack/CustomTinderCardStack";
+
+const TinderInPage = () => {
+    // const intl = useIntl();
+    return (
+        <>
```
Comment (solleOlle) [inline client/src/TinderInPage/TinderInPage.jsx:8]: why fragment?

File: client/src/_components/CustomTinderCard/CustomTinderCard.jsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+    },
+}));
+
+function CustomTinderCard({offer}) {
+    const classes = useStyles();
+    const intl = useIntl();
+    const [isLightMode] = useContext(DarkLightModeContext);
+
+    return (
+        <div className={`${styles.card} ${isLightMode ? styles.light : styles.dark}`}>
```
Comment (solleOlle) [inline client/src/_components/CustomTinderCard/CustomTinderCard.jsx:27]: we should use material UI as much as possible

File: client/src/_components/CustomTinderCard/CustomTinderCard.jsx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+                    </div>
+                    <div>
+                        <h2>{offer.title}</h2>
+                        <div className={styles.offerDiv}>
+                            <p>{offer.city}</p>
+                            <p className={styles.offerTag}>{offer.remote}</p>
+                        </div>
+                        <p>{offer.salary} <span>net/month - B2B</span></p>
+                    </div>
+                    <div className={styles.offerInfo}>
```
Comment (solleOlle) [inline client/src/_components/CustomTinderCard/CustomTinderCard.jsx:41]: why don't we manage `offerInfoTail` style here for example? there are to many duplicates in your case

File: client/src/_components/CustomTinderCard/CustomTinderCard.jsx [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+                        <div className={`${styles.offerInfoTail} ${isLightMode ? styles.light : ''}`}>
```
Comment (solleOlle) [inline client/src/_components/CustomTinderCard/CustomTinderCard.jsx:42]: you could use [classnames](https://www.npmjs.com/package/classnames)

File: client/src/_components/CustomTinderCard/CustomTinderCard.jsx [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+                            <p className={styles.offerLabel}>Company name</p>
+                        </div>
+                        <div className={`${styles.offerInfoTail} ${isLightMode ? styles.light : ''}`}>
+                            <div className={styles.offerImg}>
+                                <GroupIcon style={{ color: 'orange' }} />
+                            </div>
+                            <p>{offer.companySize}</p>
+                            <p className={styles.offerLabel}>Company size</p>
+                        </div>
+                        <div className={`${styles.offerInfoTail} ${isLightMode ? styles.light : ''}`}>
```
Comment (solleOlle) [inline client/src/_components/CustomTinderCard/CustomTinderCard.jsx:56]: you create a component with structure:
<div className={`${styles.offerInfoTail} ${isLightMode ? styles.light : ''}`}>
      <div className={styles.offerImg}>
          <DonutLargeIcon style={{ color: 'blue' }} />
      </div>
      <p>{offer.dateOfAdded}</p>
      <p className={styles.offerLabel}>Added</p>
  </div>
  
  and reuse it

File: client/src/_components/CustomTinderCard/CustomTinderCard.jsx [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+                            </div>
+                            <p>{offer.experience}</p>
+                            <p className={styles.offerLabel}>EXP. lvl</p>
+                        </div>
+                        <div className={`${styles.offerInfoTail} ${isLightMode ? styles.light : ''}`}>
+                            <div className={styles.offerImg}>
+                                <DonutLargeIcon style={{ color: 'blue' }} />
+                            </div>
+                            <p>{offer.dateOfAdded}</p>
+                            <p className={styles.offerLabel}>Added</p>
```
Comment (solleOlle) [inline client/src/_components/CustomTinderCard/CustomTinderCard.jsx:68]: should we use translate this line? :)

File: client/src/_components/CustomTinderCard/CustomTinderCard.jsx [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+                                    <Typography component="legend" readOnly className={classes.rating}>{tech.description}</Typography>
+                                </Box>
+                            );
+                        })}
+                    </div>
+                </div>
+                <div className={styles.offerDescription}>
+                    <h4>{intl.formatMessage({id: 'customTinderCard.description'})}</h4>
+                    <p>{offer.description}</p>
+                    {offer.interviewOnline ? (
```
Comment (solleOlle) [inline client/src/_components/CustomTinderCard/CustomTinderCard.jsx:89]: you could use `&&` here. so you don't need `: null`

File: client/src/_components/CustomTinderCardStack/CustomTinderCardStack.jsx [hunk 12]
```diff
@@ -0,0 +1,180 @@
+import React, { useState, useMemo } from 'react';
+import TinderCard from 'react-tinder-card';
+import * as styles from './CustomTinderCardStack.module.scss';
+import CustomTinderCard from "../CustomTinderCard/CustomTinderCard";
+import {useIntl} from "react-intl";
+
+const offers = [
```
Comment (solleOlle) [inline client/src/_components/CustomTinderCardStack/CustomTinderCardStack.jsx:7]: could you move it to `services/mock/` please?

File: client/src/_components/CustomTinderCardStack/CustomTinderCardStack.jsx [hunk 13]
```diff
@@ -0,0 +1,180 @@
...
+let offersState = offers; // This fixes issues with updating characters state forcing it to use the current state and not the state that was active when the card was created.
+
+function CustomTinderCardStack() {
+    const [offersCard, setOffersCard] = useState(offers);
+    const [lastDirection, setLastDirection] = useState();
+    const intl = useIntl();
+    const childRefs = useMemo(() => Array(offers.length).fill(0).map(i => React.createRef()), []);
+
+    const swiped = (direction, id) => {
+        // console.log('removing: ' + id)
```
Comment (solleOlle) [inline client/src/_components/CustomTinderCardStack/CustomTinderCardStack.jsx:127]: could you delete it?

File: client/src/_components/CustomTinderCardStack/CustomTinderCardStack.jsx [hunk 14]
```diff
@@ -0,0 +1,180 @@
...
+            '\n' +
+            'Why do we use it?\n' +
+            'It is a long established fact that a reader will be distracted by the readable content of a page when looking at its layout. The point of using Lorem Ipsum is that it has a more-or-less normal distribution of letters, as opposed to using \'Content here, content here\', making it look like readable English. Many desktop publishing packages and web page editors now use Lorem Ipsum as their default model text, and a search for \'lorem ipsum\' will uncover many web sites still in their infancy. Various versions have evolved over the years, sometimes by accident, sometimes on purpose (injected humour and the like).\n' +
+            '\n' +
+            '\n' +
+            'Where does it come from?\n'
+    },
+]
+
+const alreadyRemoved = [];
```
Comment (solleOlle) [inline client/src/_components/CustomTinderCardStack/CustomTinderCardStack.jsx:117]: global? fake API should be isolated. use service for that

File: client/src/_components/CustomTinderCardStack/CustomTinderCardStack.jsx [hunk 15]
```diff
@@ -0,0 +1,180 @@
...
+        alreadyRemoved.push(id);
+    }
+
+    const outOfFrame = (id) => {
+        // console.log(id + ' out of FRAME')
+        offersState = offersState.filter(character => character.id !== id);
+        setOffersCard(offersState);
+    }
+
+    const swipe = (dir) => {
```
Comment (solleOlle) [inline client/src/_components/CustomTinderCardStack/CustomTinderCardStack.jsx:138]: dir - it should be current? too complex here 🙈

File: client/src/_components/CustomTinderCardStack/CustomTinderCardStack.jsx [hunk 16]
```diff
@@ -0,0 +1,180 @@
...
+                {!lastDirection ? intl.formatMessage({ id: 'customTinderCardStack.swipeToStart' }) :
+                    lastDirection === 'up' || lastDirection === 'down' ? intl.formatMessage({ id: 'customTinderCardStack.cannotSwipe' }) :
+                        lastDirection === 'left' ? intl.formatMessage({ id: 'customTinderCardStack.reject' }) :
+                            intl.formatMessage({ id: 'customTinderCardStack.apply' })
+                }
+            </h3>
+            <div className={styles.cardContainer}>
+                {offersCard.map((offer, index) =>
+                    <TinderCard
+                        ref={childRefs[index]}
```
Comment (solleOlle) [inline client/src/_components/CustomTinderCardStack/CustomTinderCardStack.jsx:161]: why we need `ref` here?

File: client/src/_components/CustomTinderCardStack/CustomTinderCardStack.jsx [hunk 17]
```diff
@@ -0,0 +1,180 @@
...
+                        className={styles.swipe}
+                        key={offer.id}
+                        onSwipe={(dir) => swiped(dir, offer.id)}
```
Comment (solleOlle) [inline client/src/_components/CustomTinderCardStack/CustomTinderCardStack.jsx:164]: tricky naming. we have function `swipe` and `swiped`

File: client/src/_components/CustomTinderCardStack/CustomTinderCardStack.jsx [hunk 18]
```diff
@@ -0,0 +1,180 @@
...
+                        onCardLeftScreen={() => outOfFrame(offer.id)}
+                        preventSwipe={['down', 'up']}
+                    >
+                        <CustomTinderCard offer={offer}/>
+                    </TinderCard>
+                )}
+            </div>
+            <div className={styles.buttons}>
+                <button className={styles.reject} onClick={() => swipe('left')}>{intl.formatMessage({ id: 'customTinderCardStack.button.reject' })}</button>
```
Comment (solleOlle) [inline client/src/_components/CustomTinderCardStack/CustomTinderCardStack.jsx:173]: line is too long

File: client/src/_components/CustomTinderCardStack/CustomTinderCardStack.module.scss [hunk 19]
```diff
@@ -0,0 +1,131 @@
...
+
+h2, h3 {
+    color: $color-white;
+}
+
+.swipe {
+    position: absolute;
+}
+
+.cardContainer {
```
Comment (solleOlle) [inline client/src/_components/CustomTinderCardStack/CustomTinderCardStack.module.scss:49]: why camelCase?

**Pull request merged**